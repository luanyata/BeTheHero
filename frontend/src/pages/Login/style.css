.login-container {
  width: 100%;
  max-width: 1120px;
  height: 100vh;
  margin: 0 auto;

  display: flex;
  align-items: center;
  justify-content: center;
}

.login-container section.form {
  width: 100%;
  max-width: 350px;
  margin-right: 30px;
}

.login-container section.form form {
  margin-top: 100px;
}

.login-container section.form form h1 {
  font-size: 32px;
  margin-bottom: 32px;
}

.login-container section.form form a {
  display: flex;
  align-items: center;
  margin-top: 40px;
  color: #41414d;
  font-size: 18px;
  text-decoration: none;
  font-weight: 500;
  transition: opacity 0.2s;
}

.login-container section.form form a svg {
  margin-right: 8px;
}

.login-container section.form form a:hover {
  opacity: 0.8;
}
