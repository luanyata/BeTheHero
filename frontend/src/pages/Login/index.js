import React from "react";
import { Link } from "react-router-dom";
import { FiLogIn } from "react-icons/fi";
import "./style.css";

import logoImg from "../../assets/logo.svg";
import heroesImg from "../../assets/heroes.png";

function Login() {
  return (
    <div className="login-container">
      <section className="form">
        <img src={logoImg} alt="Be The Hero" />

        <form>
          <h1>Faça seu login</h1>
          <input placeholder="Sua ID" />
          <button className="button" type="submit">
            Entrar
          </button>
          <Link to="/register" className="back-link">
            <FiLogIn size={16} color="#E02041" />
            Não tenho cadastro
          </Link>
        </form>
      </section>
      <img src={heroesImg} alt="Heroes" />
    </div>
  );
}

export default Login;
