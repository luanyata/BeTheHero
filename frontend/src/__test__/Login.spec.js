import React from "react";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import Login from "../pages/Login";

describe("<Login />", () => {
  it("should to render the <Login /> component", () => {
    const wrapper = mount(
      <MemoryRouter>
        <Login />
      </MemoryRouter>
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
