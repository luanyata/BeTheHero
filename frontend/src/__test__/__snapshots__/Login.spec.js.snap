// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should to render the <Login /> component 1`] = `
"<MemoryRouter>
  <Router history={{...}}>
    <Login>
      <div className=\\"login-container\\">
        <section className=\\"form\\">
          <img src=\\"logo.svg\\" alt=\\"Be The Hero\\" />
          <form onSubmit={[Function: handleLogin]}>
            <h1>
              Faça seu login
            </h1>
            <input placeholder=\\"Login\\" value=\\"\\" onChange={[Function: onChange]} />
            <input type=\\"password\\" placeholder=\\"Senha\\" value=\\"\\" onChange={[Function: onChange]} />
            <button className=\\"button\\" type=\\"submit\\">
              Entrar
            </button>
            <Link to=\\"/register\\" className=\\"back-link\\">
              <LinkAnchor className=\\"back-link\\" href=\\"/register\\" navigate={[Function: navigate]}>
                <a className=\\"back-link\\" href=\\"/register\\" onClick={[Function: onClick]}>
                  <FiLogIn size={16} color=\\"#E02041\\">
                    <IconBase attr={{...}} size={16} color=\\"#E02041\\">
                      <svg stroke=\\"currentColor\\" fill=\\"none\\" strokeWidth=\\"2\\" viewBox=\\"0 0 24 24\\" strokeLinecap=\\"round\\" strokeLinejoin=\\"round\\" size={16} color=\\"#E02041\\" className={[undefined]} style={{...}} height={16} width={16} xmlns=\\"http://www.w3.org/2000/svg\\">
                        <path d=\\"M15 3h4a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2h-4\\" />
                        <polyline points=\\"10 17 15 12 10 7\\" />
                        <line x1=\\"15\\" y1=\\"12\\" x2=\\"3\\" y2=\\"12\\" />
                      </svg>
                    </IconBase>
                  </FiLogIn>
                  Não tenho cadastro
                </a>
              </LinkAnchor>
            </Link>
          </form>
        </section>
        <img src=\\"heroes.png\\" alt=\\"Heroes\\" />
      </div>
    </Login>
  </Router>
</MemoryRouter>"
`;
